api/src/main/java/org/cardanofoundation/tokenmetadata/registry/api/util/AssetType.java

```diff
@@ -0,0 +1,62 @@
+package org.cardanofoundation.tokenmetadata.registry.api.util;
+
+import com.bloxbean.cardano.client.util.HexUtil;
+import com.fasterxml.jackson.annotation.JsonKey;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Record/Utility class to group in one place Asset unit, policy and name manipulation and comparison
+ *
+ * @param policyId
+ * @param assetName
+ */
+@Slf4j
+public record AssetType(String policyId, String assetName) {
+
+    private static final String ADA = "ada";
+
+    private static final String LOVELACE = "lovelace";
+
+    private static final AssetType Ada = new AssetType("", LOVELACE);
+
+    public String toUnit() {
+        return policyId + assetName;
+    }
+
+    public boolean isAda() {
+        return this.equals(Ada);
+    }
+
+    @JsonKey
+    public String unsafeHumanAssetName() {
+        if (this.isAda()) {
+            return ADA;
+        } else {
+            return new String(HexUtil.decodeHexString(assetName));
+        }
+    }
+
+    public static AssetType fromUnit(String unit) {
+        try {
+            if (unit.equalsIgnoreCase(LOVELACE) || unit.trim().isEmpty()) {
+                return Ada;
```

Comment:
cosmetics: you don't need else for return, you could escape early without nested code blocks.